test/wdio.config.js

```diff
@@ -6,73 +6,117 @@ require('@babel/register')({
-  key: sauceKey,
-  capabilities: [
-    {
-      browserName: 'chrome',
-      platformName: 'Windows 10',
-      'sauce:options': {
-        build: buildNumber
-      }
-    },
-    {
-      browserName: 'firefox',
-      browserVersion: '55',
-      platformName: 'Windows 10',
-      'sauce:options': {
-        build: buildNumber
-      }
-    },
-    {
-      browserName: 'internet explorer',
-      browserVersion: '11.285',
-      platformName: 'Windows 10',
-      'sauce:options': {
-        build: buildNumber
-      }
+/**
+ * Browsers for local tests
+ *
+ * @type {RemoteCapabilities}
+ */
+const capabilitiesLocal = [
+  {
+    browserName: 'chrome',
+    'goog:chromeOptions': {
+      args: ['--headless=new'],
+      binary: puppeteer.executablePath()
     }
-  ],
-  services: services.concat([['sauce', { sauceConnect: true }]])
-}
+  }
+]
 
-const puppeteerConfig = {
-  automationProtocol: 'devtools',
-  capabilities: [
-    // { browserName: 'firefox' },
-    {
-      browserName: 'chrome',
-      'goog:chromeOptions': {
-        args: ['--headless'],
-        binary: puppeteer.executablePath()
-      }
+/**
+ * Browsers for Sauce Labs tests
+ *
+ * @type {RemoteCapabilities}
+ */
+const capabilitiesSauce = [
+  {
+    browserName: 'chrome',
+    browserVersion: 'latest',
+    platformName: 'Windows 10',
+    'sauce:options': {
+      build: SAUCE_BUILD_NUMBER
     }
-  ],
-  services
-}
+  },
+  {
+    browserName: 'firefox',
+    browserVersion: '55',
+    platformName: 'Windows 10',
+    'sauce:options': {
+      build: SAUCE_BUILD_NUMBER
+    }
+  },
+  {
+    browserName: 'internet explorer',
+    browserVersion: 'latest',
+    platformName: 'Windows 10',
+    'sauce:options': {
+      build: SAUCE_BUILD_NUMBER
+    }
+  }
+]
+
+/**
+ * WebdriverIO config
+ *
+ * @type {Testrunner}
+ */
+exports.config = {
+  user: SAUCE_USERNAME,
+  key: SAUCE_ACCESS_KEY,
+
+  // Use DevTools prototype for Puppeteer
+  automationProtocol: SAUCE_ENABLED === 'true'
+    ? 'webdriver'
+    : 'devtools',
+
+  baseUrl: `http://localhost:${PORT}`,
 
-exports.config = Object.assign({
+  capabilities: SAUCE_ENABLED === 'true'
+    ? capabilitiesSauce
+    : capabilitiesLocal,
+
+  framework: 'mocha',
   outputDir: join(cwd(), 'logs'),
-  specs: [join(cwd(), 'test/integration/**/*.js')],
-  baseUrl: `http://localhost:${staticServerPort}`,
   reporters: ['spec'],
-  framework: 'mocha',
-  mochaOpts: { timeout: 30 * 1000 }
-}, sauceEnabled ? sauceConfig : puppeteerConfig)
+
+  services: [
+    /**
+     * Web server options
+     *
+     * @type {[string, StaticServerOptions]}
+     */
+    ['static-server', {
+      folders: [
+        { mount: '/', path: join(cwd(), 'examples') },
+        { mount: '/dist/', path: join(cwd(), 'dist') }
+      ],
+      port: PORT
+    }],
+
+    /**
+     * Web server options
+     *
+     * @type {[string, SauceServiceConfig]}
+     */
+    ['sauce', {
+      sauceConnect: SAUCE_ENABLED === 'true'
+    }]
+  ],
```

Comment:
**suggestion** This part tripped me up a bit, as it always sets up a [Sauce service](https://webdriver.io/docs/sauce-service), even when testing without the `SAUCE_...` environment variables. This makes me a bit uneasy, as I could only _think_ that it was not affecting any local tests through:
- `sauceConnect` would be false due to the lack of `SAUCE_ENABLED`
- the lack of `SAUCE_USERNAME` and `SAUCE_ACCESS_KEY` environment variables to configure the connection to SauceLabs
- the lack of `sauce:options` in the `localCapabilities` (less sure of that one)

I'd find it safer to not set up the service altogether (unfortunately, requires a `.filter(Boolean) to clean up the `null` value):
```suggestion
    /**
     * WebdriverIO Sauce service options
     *
     * @type {[string, SauceServiceConfig]}
     */
    SAUCE_ENABLED === 'true'
      ? ['sauce', { sauceConnect: true }]
      : null
  ].filter(Boolean)
```

Alternatively, I'd be fine with a comment in the code to explain why this doesn't impact local runs.

Reply:
Brill, I'll add a comment to say we're using `sauceConnect` as designed

E.g. I changed it after seeing an example use `sauceConnect` to optionally enable or disable Sauce Labs